modules/identity/src/Volo.Abp.Identity.Domain/Volo/Abp/Identity/IdentitySecurityLogHandler.cs

```diff
@@ -0,0 +1,86 @@
+﻿using System;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Identity;
+using Volo.Abp.DependencyInjection;
+using Volo.Abp.EventBus;
+using Volo.Abp.Security.Claims;
+using Volo.Abp.SecurityLog;
+using Volo.Abp.Uow;
+using Volo.Abp.Users;
+
+namespace Volo.Abp.Identity
+{
+    public class IdentitySecurityLogHandler : ILocalEventHandler<IdentitySecurityLogEvent>, ITransientDependency
+    {
+        protected ISecurityLogManager SecurityLogManager { get; }
+        protected IdentityUserManager UserManager { get; }
+        protected ICurrentPrincipalAccessor CurrentPrincipalAccessor { get; }
+        protected IUserClaimsPrincipalFactory<IdentityUser> UserClaimsPrincipalFactory { get; }
+        protected ICurrentUser CurrentUser { get; }
+
+        public IdentitySecurityLogHandler(
+            ISecurityLogManager securityLogManager,
+            IdentityUserManager userManager,
+            ICurrentPrincipalAccessor currentPrincipalAccessor,
+            IUserClaimsPrincipalFactory<IdentityUser> userClaimsPrincipalFactory,
+            ICurrentUser currentUser)
+        {
+            SecurityLogManager = securityLogManager;
+            UserManager = userManager;
+            CurrentPrincipalAccessor = currentPrincipalAccessor;
+            UserClaimsPrincipalFactory = userClaimsPrincipalFactory;
+            CurrentUser = currentUser;
+        }
+
+        public async Task HandleEventAsync(IdentitySecurityLogEvent eventData)
+        {
+            Action<SecurityLogInfo> securityLogAction = securityLog =>
+            {
+                securityLog.Identity = eventData.Identity;
+                securityLog.Action = eventData.Action;
+
+                if (securityLog.UserName.IsNullOrWhiteSpace())
+                {
+                    securityLog.UserName = eventData.UserName;
+                }
+
+                if (securityLog.ClientId.IsNullOrWhiteSpace())
```

Comment:
Shouldn't be vice verse? I mean a filled eventData.ClientId should override the securityLog.ClientId. What do you think?
I didn't think much, but it seems like that. If that's true, no problem.